Author: Guo Yixuan <culu.gyx@gmail.com>
Description: Better clean target for several Makefiles
 This would save some rm's in override_dh_auto_clean.
--- a/samples/sleeper/Makefile
+++ b/samples/sleeper/Makefile
@@ -19,11 +19,10 @@
 libstdc++.a:
 	ln -s `g++ -print-file-name=libstdc++.a`
 
-clean:
-	rm $(PROGS)
+clean: distclean
 
 distclean:
-	/bin/rm -f $(PROGS) *.o libstdc++.a
+	rm -f $(PROGS) *.o libstdc++.a
 
 sleeper: sleeper.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a
 	g++ $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o sleeper sleeper.o libstdc++.a -pthread -lboinc_api -lboinc
--- a/samples/worker/Makefile
+++ b/samples/worker/Makefile
@@ -4,11 +4,10 @@
 
 all: $(PROGS)
 
-clean:
-	rm $(PROGS)
+clean: distclean
 
 distclean:
-	rm $(PROGS)
+	rm -f $(PROGS) *.o
 
 worker: worker.o
 	g++ $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o worker worker.o
--- a/samples/wrapper/Makefile
+++ b/samples/wrapper/Makefile
@@ -24,11 +24,10 @@
 libstdc++.a:
 	ln -s `g++ -print-file-name=libstdc++.a`
 
-clean:
-	rm $(PROGS) *.o
+clean: distclean
 
 distclean:
-	/bin/rm -f $(PROGS) *.o libstdc++.a
+	rm -f $(PROGS) *.o libstdc++.a
 
 REGEXP_OBJS = \
 	regexp.o \
